### PR TITLE
Use Span.IndexOf rather than Contains

### PR DESCRIPTION
**Pull request by benaadams**
It will only search for the byte if its found; so shouldn't be slower in fast path

/cc @davidfowl

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/HttpParser.cs [hunk 1]
```diff
@@ -393,50 +399,16 @@ private unsafe void TakeSingleHeader(byte* headerLine, int length, IHttpHeadersH
                 {
                     break;
                 }
+
+                valueLength--;
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/HttpParser.cs]: Put this in the for loop.

File: src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/HttpParser.cs [hunk 2]
```diff
@@ -69,6 +69,8 @@ public unsafe bool ParseRequestLine(IHttpRequestLineHandler handler, ReadableBuf
 
         private unsafe void ParseRequestLine(IHttpRequestLineHandler handler, byte* data, int length)
         {
+            Debug.Assert(length >= 0);
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/HttpParser.cs]: Remove the asserts

File: src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/HttpParser.cs (hunk 1 above)
Comment (benaadams) [inline src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/HttpParser.cs]: Changed to bool; and resuse span if not changed

File: src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/HttpParser.cs (hunk 2 above)
Comment (benaadams) [inline src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/HttpParser.cs]: Removed

File: src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/HttpParser.cs [hunk 3]
```diff
@@ -379,62 +378,35 @@ private unsafe void TakeSingleHeader(byte* headerLine, int length, IHttpHeadersH
             }
 
             // Check for CR in value
-            var i = valueStart + 1;
-            if (Contains(headerLine + i, valueEnd - i, ByteCR))
+            var valueLength = valueEnd - valueStart + 1;
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/HttpParser.cs]: Get rid of this local, it isn't kept up to date.

**Pull request merged**